### PR TITLE
Update nuspecs for netstandard1.3 change

### LINKED ISSUE #14248: Microsoft.Net.Compilers package is broken due to missing Facades.
The Microsoft.Net.Compilers package is currently broken due to missing Facades. Attempting to build using the latest Microsoft.Net.Compilers package on MyGet currently results in the following:

```

Unhandled Exception: System.IO.FileNotFoundException: Could not load file or assembly 'System.Diagnostics.FileVersionInfo, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
   at Microsoft.CodeAnalysis.CommonCompiler.GetAssemblyFileVersion()
   at Microsoft.CodeAnalysis.CSharp.CSharpCompiler.PrintLogo(TextWriter consoleOutput)
   at Microsoft.CodeAnalysis.CommonCompiler.RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CommonCompiler.Run(TextWriter consoleOutput, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CommandLine.Csc.<>c__DisplayClass1_0.<Run>b__0(TextWriter tw)
   at Microsoft.CodeAnalysis.CSharp.CommandLine.Csc.Run(String[] args, BuildPaths buildPaths, TextWriter textWriter, IAnalyzerAssemblyLoader analyzerLoader)
   at Microsoft.CodeAnalysis.CommandLine.DesktopBuildClient.RunLocalCompilation(String[] arguments, BuildPaths buildPaths, TextWriter textWriter)
   at Microsoft.CodeAnalysis.CommandLine.BuildClient.RunCompilation(IEnumerable`1 originalArguments, BuildPaths buildPaths, TextWriter textWriter)
   at Microsoft.CodeAnalysis.CommandLine.DesktopBuildClient.Run(IEnumerable`1 arguments, IEnumerable`1 extraArguments, RequestLanguage language, CompileFunc compileFunc, IAnalyzerAssemblyLoader analyzerAssemblyLoader)
   at Microsoft.CodeAnalysis.CSharp.CommandLine.Program.Main(String[] args, String[] extraArgs)
   at Microsoft.CodeAnalysis.CSharp.CommandLine.Program.Main(String[] args)
```

This is currently blocking one of our repositories.

### PR DESCRIPTION
**Pull request by agocke**
Fixes #14248 

/cc @jaredpar @tannergooding @jasonmalinowski  @dotnet/roslyn 

Someone on the IDE team may want to look at the changes to the Workspaces/Features nuspecs and validate that they're expected.

### CODE REVIEW

**Comment by tannergooding** [comment]
LGTM (modulo `System.Collections.Immutable` and `System.Reflection.Metadata` needing to be added back to the `Microsoft.Net.Compilers` nuspec).

**Comment by shyamnamboodiripad** [comment]
@jinujoseph (this impacts nugets) @tmat

**Comment by jaredpar** [comment]
This breaks all NuGet v2 clients correct?

**Comment by agocke** [comment]
@jaredpar I would expect so -- they don't know about this TFM.

**Comment by tannergooding** [comment]
@jaredpar, @agocke. I'm pretty sure the latest v2.8 NuGet supports the NetStandard TFMs (but only the latest).

**Comment by jaredpar** [comment]
Can we get confirmation before merging?

**Comment by tannergooding** [comment]
@jaredpar, @agocke... Looks like the command line itself for v2 doesn't support NetStandard (at least as far as I can tell from the Release Notes). However, the VS driver does: http://docs.nuget.org/ndocs/release-notes/nuget-2.12.

So the only scenario that would be broken here (confirming now) is that a user that explicitly downloads the v2.8.6 command line executable (or prior) and tries to do a command line restore/install, will have it fail.

**Comment by tannergooding** [comment]
Actually, I'm not seeing any issues doing a restore or install with a `netstandard1.3` only package (this is using the nuget v2.8.6 command line utility and packages.config).

**Comment by tannergooding** [comment]
Someone else should probably independently validate my findings though

**Comment by jaredpar** [comment]
CC @jasonmalinowski to see if he knows

**Comment by Pilchie** [comment]
There are a couple of IDE bugs about the nuspecs that @shyamnamboodiripad filed - does this fix any of them?

**Comment by agocke** [comment]
@Pilchie No.

**Comment by shyamnamboodiripad** [comment]
(For reference, I think the bugs @Pilchie referred to above are #12444 and #12017)

File: src/NuGet/Microsoft.CodeAnalysis.Workspaces.Common.nuspec [hunk 1]
```diff
@@ -27,12 +27,8 @@ Supported Platforms:
     <serviceable>true</serviceable>
   </metadata>
   <files>
-    <file src="Microsoft.CodeAnalysis.Workspaces.dll" target="lib\portable-net45+win8" />
-    <file src="Microsoft.CodeAnalysis.Workspaces.xml" target="lib\portable-net45+win8" />
-    <file src="Microsoft.CodeAnalysis.Workspaces.dll" target="lib\net45" />
-    <file src="Microsoft.CodeAnalysis.Workspaces.xml" target="lib\net45" />
-    <file src="Microsoft.CodeAnalysis.Workspaces.Desktop.dll" target="lib\net45" />
-    <file src="Microsoft.CodeAnalysis.Workspaces.Desktop.xml" target="lib\net45" />
+    <file src="Microsoft.CodeAnalysis.Workspaces.Desktop.dll" target="lib\net46" />
```
Comment (jasonmalinowski) [inline src/NuGet/Microsoft.CodeAnalysis.Workspaces.Common.nuspec]: There should also be a copy of Microsoft.CodeAnalysis.Workspaces.dll in the lib\net46 target as well. The .Desktop API is additive, not a different implementation of the same. #12444 is tracking a further cleanup, but you'll want to do that minimally.

File: src/NuGet/Microsoft.CodeAnalysis.Workspaces.Common.nuspec [hunk 2]
```diff
@@ -27,12 +27,8 @@ Supported Platforms:
     <serviceable>true</serviceable>
   </metadata>
```
Comment (jasonmalinowski) [inline src/NuGet/Microsoft.CodeAnalysis.Workspaces.Common.nuspec]: Does the dependencies list need to be updated as well? There's a bunch of other packages we're using too in src\Workspaces\Core\Portable\project.json:

```
    "System.Diagnostics.Contracts": "4.0.1",
    "System.Linq.Parallel": "4.0.1",
    "System.ObjectModel": "4.0.10",
    "System.Text.RegularExpressions": "4.1.0",
    "System.Threading.Tasks.Parallel": "4.0.1"
```

**Comment by agocke** [comment]
@jasonmalinowski Fixed

**Comment by tannergooding** [comment]
@agocke, `minClientVersion` needs to be specified like `<metadata minClientVersion="3.3">`.

**Comment by agocke** [comment]
@jasonmalinowski Updated

**Comment by tannergooding** [comment]
Can this be merged now? It is needed for us to move to the latest bits 😄

**Comment by agocke** [comment]
@jasonmalinowski You've marked changes requested. I'd like you to explicitly sign off on this, please.

**Comment by ManishJayaswal** [comment]
Yes this is blocking us from inserting our bits into RC. We would appreciate early merge of this.

**Comment by Pilchie** [comment]
NOTE: @jasonmalinowski will be offline for at about 1 more hour...

**Comment by agocke** [comment]
@Pilchie Would you sign off instead? The changes impact the IDE so I'd like explicit approval from someone on the IDE team.

**Comment by tannergooding** [comment]
ping @Pilchie or @jasonmalinowski for review.

File: src/NuGet/Microsoft.CodeAnalysis.Common.nuspec [hunk 3]
```diff
@@ -1,23 +1,12 @@
...
-      - Portable Class Libraries
     </description>
     <dependencies>
-      <group targetFramework="portable-net45+win8">
-        <dependency id="System.Collections.Immutable"                  version="$SystemCollectionsImmutableVersion$" />
-        <dependency id="System.Reflection.Metadata"                    version="$SystemReflectionMetadataVersion$" />
-        <dependency id="Microsoft.CodeAnalysis.Analyzers"              version="$MicrosoftCodeAnalysisAnalyzersVersion$" />
-      </group>
-
       <group targetFramework="netstandard1.3">
```
Comment (Pilchie) [inline src/NuGet/Microsoft.CodeAnalysis.Common.nuspec]: Nit: We could get rid of the "group" now, right?

Comment (Pilchie) [inline src/NuGet/Microsoft.CodeAnalysis.Common.nuspec]: Also, this change makes it so ugly for people using packages.config :(

File: src/NuGet/Microsoft.CodeAnalysis.Compilers.nuspec [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
 <package xmlns="http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd">
-  <metadata>
+  <metadata minClientVersion="3.3">
     <id>Microsoft.CodeAnalysis.Compilers</id>
     <summary>.NET Compiler Platform ("Roslyn")</summary>
     <description>
       .NET Compiler Platform ("Roslyn"). Install this package to get both C# and Visual Basic support. Install either of the dependencies directly to get one of the languages separately.
-          
+
 Supported Platforms:
```
Comment (Pilchie) [inline src/NuGet/Microsoft.CodeAnalysis.Compilers.nuspec]: Not updated?

**Comment by tannergooding** [comment]
I'm going to merge this, now that we have sign-off, so that we are unblocked from inserting. The two remaining fixes (removing the group and updated the Support Platforms list) are fairly simple and can be handled in a follow up PR.

**Comment by jasonmalinowski** [comment]
@tannergooding/@agocke: what about my other question about the Workspace.Common dependency list?

**Comment by tannergooding** [comment]
@jasonmalinowski, it was already fixed?

**Comment by tannergooding** [comment]
https://github.com/agocke/roslyn/commit/623a53a07b247949611f6d7b8c56952e089a0f9b

**Comment by jasonmalinowski** [comment]
Oops, sorry. Reading is hard.

**Comment by jaredpar** [comment]
> We should update all of the .nuspecs to have the minimum client version set so downlevel clients don't blow up catastrophically.

This sounds like we're deliberately cutting off support for 2.0 clients.  Is that really the case?

**Comment by agocke** [comment]
@jaredpar They'll fail anyway, I think this just makes the error nicer. Now they know they can just go update their NuGet exe and everything will work.

**Comment by jaredpar** [comment]
That's unfortunate but I understand.

**Pull request merged**